composer/trainer/trainer.py

```diff
@@ -945,21 +911,23 @@ def __init__(
         # Callbacks
         self.state.callbacks[:] = list(cast(List[Callback], loggers)) + self.state.callbacks
 
+        checkpoint_callbacks = [callback for callback in self.state.callbacks if isinstance(callback, CheckpointSaver)]
         # Checkpoint Saving
+
         self._checkpoint_saver = None
+        # If the user specifies Checkpointing arguments in the Trainer, then create
+        # a _checkpoint_saver callback from their args.
         if checkpoint_save_path is not None:
             self._checkpoint_saver = CheckpointSaver(
                 checkpoint_save_path=checkpoint_save_path,
-                checkpoint_filename=save_filename,
-                artifact_name=save_artifact_name,
-                latest_checkpoint_filename=save_latest_filename,
-                latest_artifact_name=save_latest_artifact_name,
-                overwrite=save_overwrite,
-                weights_only=save_weights_only,
                 checkpoint_save_interval=checkpoint_save_interval,
                 num_checkpoints_to_keep=num_checkpoints_to_keep,
             )
             self.state.callbacks.append(self._checkpoint_saver)
+        # If the user specified a checkpoint callback, then use that as the _checkpoint_saver callback.
+        # TODO(eracah): deal with multiple CheckpointSaver callbacks.
```

Comment:
Will this be fixed in current series of PRs? Or do we need to move to JIRA?

Reply:
oh yeah let me move to JIRA

Reply:
done

Reply:
Thanks for filing the Jira to track!  Can we remove the TODO and print a warning here notifying the user that we're just using the first callback they specified in the list?

Reply:
done!